Desktop_Interface/functiongencontrol.h

```diff
@@ -29,39 +30,66 @@ struct ChannelData
 	double offset = 0.0;
 };
 
-class functionGenControl : public QLabel
+class SingleChannelController : public QObject
 {
-    Q_OBJECT
+	Q_OBJECT
+
 public:
-    explicit functionGenControl(QWidget *parent = 0);
+	ChannelData const& getData() const;
+
+signals:
+    void notifyUpdate(SingleChannelController* controller);
+	void setMaxFreq(double maxFreq);
+    void setMinFreq(double minFreq);
+
+public slots:
+    void waveformName(QString newName);
+    void freqUpdate(double newFreq);
+    void amplitudeUpdate(double newAmplitude);
+    void offsetUpdate(double newOffset);
 
-	ChannelData channels[2];
+private:
+	ChannelData m_data;
+};
 
+class DualChannelController : public QLabel
+{
+    Q_OBJECT
 public:
+    explicit DualChannelController(QWidget *parent = 0);
 
-	ChannelData& getChannelData(ChannelID channelID);
+public:
+	SingleChannelController* getChannelController(ChannelID channelID);
 
 signals:
-    void functionGenToUpdate(ChannelID channel, ChannelData *fGenControlData);
+    void functionGenToUpdate(ChannelID channel, SingleChannelController* fGenControl);
     void setMaxFreq_CH1(double maxFreq);
     void setMinFreq_CH1(double minFreq);
     void setMaxFreq_CH2(double maxFreq);
     void setMinFreq_CH2(double minFreq);
 
 public slots:
-    void waveformName(ChannelID channelId, QString newName);
-    void freqUpdate(ChannelID channelId, double newFreq);
-    void amplitudeUpdate(ChannelID channelId, double newAmplitude);
-    void offsetUpdate(ChannelID channelId, double newOffset);
+    void waveformName(ChannelID channelID, QString newName);
+    void freqUpdate(ChannelID channelID, double newFreq);
+    void amplitudeUpdate(ChannelID channelID, double newAmplitude);
+    void offsetUpdate(ChannelID channelID, double newOffset);
 
     void waveformName_CH1(QString newName);
     void freqUpdate_CH1(double newFreq);
     void amplitudeUpdate_CH1(double newAmplitude);
     void offsetUpdate_CH1(double newOffset);
+
     void waveformName_CH2(QString newName);
     void freqUpdate_CH2(double newFreq);
     void amplitudeUpdate_CH2(double newAmplitude);
     void offsetUpdate_CH2(double newOffset);
+
+private:
+	SingleChannelController m_channels[2];
 };
 
+}
+
+using functionGenControl = functionGen::DualChannelController;
```

Comment:
Understand why you did this (makes testing a whole lot easier), but as a permanent solution there's a little bit of a smell there.

Reply:
Of course, I just didn't want this diff to force me to change many different files unnecesarily